tests/debug-mode/host/host.c

```diff
@@ -68,10 +68,6 @@ static void _test_debug_signed(const char* path)
 {
     /* Signed debug mode should always pass. */
     _launch_enclave_success(path, _create_flags(true));
-#ifdef OE_USE_LIBSGX
```

Comment:
This removal reduces our test coverage. The problem with this test as originally implemented is that it used `OE_USE_LIBSGX` as a signal for FLC support, which is not served by the new DCAP flag. This test needs to do a runtime detection now:

- It can perform a CPUID check for the presence of FLC and only execute the non-debug enclave launch test if found.
    - The problem with this is that we may be running the AESM configuration on an FLC capable machine. It would be worth syncing with @johnkord to see if this is a configuration we run.

- Add a new `#ifdef` that is dependent on the new DCAP definition. i.e. building for the DCAP configuration requires that FLC be present, so it can still be used as a signal, even though it is not the exact semantic requirement.

Reply:
`OE_HAS_QUOTE_PROVIDER` was meant to be a tell whether we have FLC or not. Now it's renamed to `OE_LINK_SGX_DCAP_QL`, but should have the same semantic.

Reply:
The original comments says the deleted line only works for NGSA SDK. I assume that's not a true statement and needs to be updated?

Reply:
Yes, the comment needs to be updated. NGSA SDK is a prerelease name for the DCAP SDK, so they mean the same thing if you're using the `OE_LINK_SGX_DCAP_QL` define now

Reply:
Added the line back and updated comments.